src/main/java/nextstep/subway/domain/Section.java

```diff
@@ -0,0 +1,130 @@
+package nextstep.subway.domain;
+
+import java.util.Arrays;
+import java.util.List;
+import java.util.Objects;
+
+import javax.persistence.Entity;
+import javax.persistence.FetchType;
+import javax.persistence.GeneratedValue;
+import javax.persistence.GenerationType;
+import javax.persistence.Id;
+import javax.persistence.JoinColumn;
+import javax.persistence.ManyToOne;
+
+@Entity
+public class Section {
+    private static final String DISTANCE_OVER_ERROR_MESSAGE = "추가할 구간의 거리가 기존 역 사이 거리보다 길 수는 없습니다.";
+
+    @Id
+    @GeneratedValue(strategy = GenerationType.IDENTITY)
+    private Long id;
+
+    @ManyToOne(fetch = FetchType.LAZY)
+    @JoinColumn(name = "line_id")
+    private Line line;
+
+    @ManyToOne(fetch = FetchType.LAZY)
+    @JoinColumn(name = "up_station_id")
+    private Station upStation;
+
+    @ManyToOne(fetch = FetchType.LAZY)
+    @JoinColumn(name = "down_station_id")
+    private Station downStation;
+
+    private Long distance;
+
+    protected Section() {
+    }
+
+    public Section(Long id, Line line, Station upStation, Station downStation, Long distance) {
+        this.id = id;
+        this.line = line;
+        this.upStation = upStation;
+        this.downStation = downStation;
+        this.distance = distance;
+    }
+
+    public Section(Long id, Station upStation, Station downStation, Long distance) {
+        this(id, null, upStation, downStation, distance);
+    }
+
+    public Section(Line line, Station upStation, Station downStation, Long distance) {
+        this(null, line, upStation, downStation, distance);
+    }
+
+    public Section(Station upStation, Station downStation, Long distance) {
+        this(null, null, upStation, downStation, distance);
+    }
+
+    public void modify(Section section) {
+        modifyUpStation(section);
+        modifyDownStation(section);
+    }
+
+    private void modifyUpStation(Section section) {
+        if (!Objects.equals(this.upStation, section.upStation)) {
+            return;
+        }
+        this.upStation = section.downStation;
+        modifyDistance(section.distance);
+    }
+
+    private void modifyDownStation(Section section) {
+        if (!Objects.equals(this.downStation, section.downStation)) {
+            return;
+        }
+        this.downStation = section.upStation;
+        modifyDistance(section.distance);
+    }
+
+    private void modifyDistance(Long distance) {
+        validateDistanceOver(distance);
+        this.distance = this.distance - distance;
+    }
+
+    private void validateDistanceOver(Long distance) {
+        if (distance >= this.distance) {
+            throw new IllegalArgumentException(DISTANCE_OVER_ERROR_MESSAGE);
+        }
+    }
```

Comment:
```suggestion
    private void validateDistanceOver(Distance distance) {
        if(distance.greaterThan(distance)) {
            ....
```

`Distance` 객체에게 시켜야될 메시지가 몇가지 보이는데요 책임에 맞게 분리해본다면 어떨까요?

Reply:
Distance를 원시 타입 포장 객체로 분리하겠습니다!